plotter/colorbar.go

```diff
@@ -0,0 +1,107 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"image"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/palette"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// ColorBar is a plot.Plotter that draws a color bar legend for a ColorMap.
+type ColorBar struct {
+	// Vertical determines wether the legend will be
+	// plotted vertically or horizontally.
+	// The default is false (horizontal).
+	Vertical bool
+
+	// Colors specifies the number of colors to be
+	// shown in the legend. The default is 255.
+	Colors int
+
+	cm palette.ColorMap
+}
+
+// NewColorBar creates a new ColorBar plotter.
+func NewColorBar(cm palette.ColorMap) *ColorBar {
+	return &ColorBar{
+		cm:     cm,
+		Colors: 255,
+	}
+}
+
+// Plot implements the Plot method of the plot.Plotter interface.
+func (l *ColorBar) Plot(c draw.Canvas, p *plot.Plot) {
+	if l.cm.Max() == l.cm.Min() {
+		panic("palette: ColorMap Max==Min")
```

Comment:
perhaps check this invariant inside `NewColorBar` instead of inside each and every method of `ColorBar`.